File: src/main/java/dev/orion/users/application/usecases/DeleteUserImpl.java
/**
 * @License
 * Copyright 2024 Orion Services @ https://orion-services.dev
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package dev.orion.users.application.usecases;

import dev.orion.users.application.interfaces.DeleteUser;

public class DeleteUserImpl implements DeleteUser {

    /**
     * Deletes a User from the service.
     *
     * @param email : User email
     *
     * @return Return 1 if user was deleted
     */
    @Override
    public boolean deleteUser(final String email) {
        if (email.isBlank()) {
            throw new IllegalArgumentException("Email can not be blank");
        } else {
            return true;
        }
    }

}

File: src/main/kotlin/dev/orion/users/application/usecases/DeleteUserImpl.kt
/**
 * @License
 * Copyright 2025 Orion Services @ https://orion-services.dev
 */
package dev.orion.users.application.usecases

import dev.orion.users.application.port.`in`.DeleteUser
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject

@ApplicationScoped
open class DeleteUserImpl @Inject constructor() : DeleteUser {

    override fun deleteUser(email: String): Boolean {
        if (email.isBlank()) {
            throw IllegalArgumentException("Email can not be blank")
        }
        return true
    }
}
